src/main/java/com/android/volley/toolbox/DiskBasedCache.java

```diff
@@ -241,8 +236,10 @@ public synchronized void put(String key, Entry entry) {
             fos.write(entry.data);
             fos.close();
             e.size = file.length();
-            putEntry(key, e);
-            pruneIfNeeded();
+            mTotalSize = DiskBasedCacheUtility.putEntry(key, e, mTotalSize, mEntries);
```

Comment:
remove "mTotalSize = " - no need to set this twice, and the parent method and I believe all other reads of mTotalSize are synchronized so mTotalSize shouldn't be observable between these two sets.

Reply:
Disagree. The updated value of mTotalSize is passed to pruneIfNeeded. (I hope a unit test would catch this...)

By the way, having to pass all of these values to the tool method, in the right order, and then update the instance fields, seems brittle. Can't OOP offer more assistance more here?

Reply:
but it is passed into the next method, so doesn't it need to be updated there?

Reply:
Yeah, I was thinking about better ways to do this. I think it's good to share the code, since it'd be a lot of code to be duplicated. I don't think this is terrible, since the methods take in all different types, which means it'll tell you if the arguments aren't in the right order. The only other method I could think of is passing in the Cache itself. And then making those variables accessible. But I think that may cause more issues.

Reply:
Yeah, sorry, didn't realize it was passed from the first method to the second. Never mind here.